rules/experimental/rbe/README.md

```diff
@@ -0,0 +1,256 @@
+
+In the WORKSPACE file, call
+```
+rbe_exec_properties(
+    name = "exec_properties",
+)
+```
+
+This creates a local repo @rbe_exec_properties with standard RBE execution property constants. For
+example, NETWORK_ON which is the dict {"dockerNetwork" : "standard"}. (For the full list of these
+constants see STANDARD_PROPERTY_SETS in exec_properties.bzl.)
+
+Then, in some BUILD file in your work space, you can reference this execution property constant as
+follows:
+```
+load("@exec_properties//:constants.bzl", "NETWORK_ON")
+...
+foo_test(
+   ...
+   exec_properties = NETWORK_ON,
+)
+```
+
+### Use case 2 - local execution
+
+If Bazel is set up so that the targets are executed locally, then the contents of exec_properties
+are ignored.
+
+### Use case 3 - non-RBE remote execution:
+
+Let's assume that the non-RBE remote execution endpoint provides a macro similar to
+rbe_exec_properties (say other_re_exec_properties), which populates the same constants (e.g.
+NETWORK_ON) with possibly different dict values.
+In this case, the WORKSPACE would, presumably, look like this:
+```
+other_re_exec_properties(
+    name = "exec_properties",
+)
+```
+
+And the targets in the BUILD files will be able to depend on targets from other repos that were
+written with RBE in mind, as the name of the repo defined in the WORKSPACE (exec_properties in this
+case) is the same. That is why the repo name exec_properties does *not* contain the word RBE.
+
+### Use case 4 - rbe_exec_properties with override_constants:
+
+Let's now assume that a particular repo, running with a particular RBE setup, wants to run
+all its remote execution actions without network access, possibly for the sake of identifying any
+network dependencies. This would be achieved as follows.
+
+In the WORKSPACE file, call
+```
+rbe_exec_properties(
+    name = "exec_properties",
+    override_constants = {
+        "NETWORK_ON": create_exec_properties_dict(docker_network = "off"),
+    },
+)
+```
+
+This would override the meaning of NETWORK_ON for this workspace only.
+For this override to work, we depend on targets marking their network dependecy by using NETWORK_ON
+that was loaded from the repo @exec_properties.
+
+### Use case 5 - custom execution properties
+
+In this scenario, let's assume that a target is best run remotely on a high memory GCE machine.
+Let's also assume that the RBE setup associated with the workspace where the target is defined has
+workers of type "n1-highmem-8".
+
+Setting exec_properties = {"gceMachineType" : "n1-highmem-8"} is problematic because it does not
+lend itself to another repo depending on this target in all cases. See
+[anti-patterns](#anti-patterns) below. Unlike the case of NETWORK_ON, rbe_exec_properties does not
+provide a standard HIGH_MEM_MACHINE execution property constant (although it might do so in the
+future).
+
+The recommended way for a repo (let's call this repo, repo A) to define this high-mem dependency is
```

Comment:
use consistent naming, either Bazel project (or repo, though I think Bazel project is better)

Reply:
done